src/helpers/withLibcore.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+
+const core = require('@ledgerhq/ledger-core')
+
+let walletPool = null
+let queue = Promise.resolve()
+
+// TODO: `core` and `NJSWalletPool` should be typed
+type Job = (Object, Object) => any
+
+export default function withLibcore(job: Job) {
+  if (!walletPool) {
+    walletPool = core.instanciateWalletPool({
+      // sqlite files will be located in the app local data folder
+      dbPath: process.env.LEDGER_LIVE_SQLITE_PATH,
+    })
+  }
+  // $FlowFixMe WTF is happening here, dudes.
+  queue = queue.then(async () => {
+    try {
+      if (!walletPool) {
+        throw new Error('wallet pool not instanciated. this should not happen')
+      }
+      return job(core, walletPool)
+    } catch (e) {
+      console.log(`withLibCore: Error in job`, e) // eslint-disable-line no-console
+      return Promise.resolve()
```

Comment:
mmh to satisfy the type shared above you might want to change a bit. maybe this:

```js
const p = queue.then(() => {
  invariant(walletPool, 'wallet pool not instanciated. this should not happen')
  return job(core, walletPool)
})

queue = p.catch(() => {});

return p
```

Reply:
in fact the check is totally useless because [it is impossible that we get to this line without wallet pool instanciated](https://github.com/meriadec/ledger-live-desktop/blob/49959943256207894550eae1a1568fbfbae6b1ef/src/helpers/withLibcore.js#L12-L17), but he is here only because flow can't see it. /shrug

Reply:
a check is never useless ;P

I think what's happening here is as you have defined walletPool as null, it will be infered as `any` because `instanciateWalletPool` is not typed and flow can't tell if it does not return a null too.

Reply:
you can probably have the invariant before the `const p =` . anyway, flow is weird at first, but everything is perfectly logic as soon as you inspect what it's doing. like what type each variable was inferred too. that's why it's quite interesting to have an IDE like Nuclide to see the inferred type & also the type coverage